staging/src/k8s.io/apimachinery/pkg/util/net/port_range.go

```diff
@@ -59,19 +58,39 @@ func (pr *PortRange) Set(value string) error {
 	}
 
 	hyphenIndex := strings.Index(value, "-")
-	if hyphenIndex == -1 {
-		return fmt.Errorf("expected hyphen in port range")
+	plusIndex := strings.Index(value, "+")
+
+	// exclusive or
+	if (hyphenIndex != -1) == (plusIndex != -1) {
+		return fmt.Errorf("unable to parse port range: %s", value)
 	}
 
 	var err error
 	var low int
 	var high int
-	low, err = strconv.Atoi(value[:hyphenIndex])
-	if err == nil {
-		high, err = strconv.Atoi(value[hyphenIndex+1:])
+
+	if hyphenIndex != -1 {
+		low, err = strconv.Atoi(value[:hyphenIndex])
+		if err == nil {
+			high, err = strconv.Atoi(value[hyphenIndex+1:])
+		}
+		if err != nil {
+			return fmt.Errorf("unable to parse port range: %s: %v", value, err)
+		}
 	}
-	if err != nil {
-		return fmt.Errorf("unable to parse port range: %s: %v", value, err)
+	if plusIndex != -1 {
+		low, err = strconv.Atoi(value[:plusIndex])
+		var offset int
+		if err == nil {
+			offset, err = strconv.Atoi(value[plusIndex+1:])
+		}
+		if err != nil {
+			return fmt.Errorf("unable to parse port range: %s: %v", value, err)
+		}
+		if offset <= 0 {
```

Comment:
Remove this test. This is impossible to hit because of the xor above (would require that the string has a + and a -) and we double check it below when we test high > low.